### ISSUE TITLE
Still Login Redirect Loop after upgrade to 1.4.1, if previously cached tokens exists in browser store

### ISSUE THREAD

**Issue by kumarunster** [AzureAD/microsoft-authentication-library-for-js#2370]
## Library
- [x ] `msal@1.4.1`
- [ ] `@azure/msal-browser@2.x.x`
- [ ] `@azure/msal-angular@0.x.x`
- [ ] `@azure/msal-angular@1.x.x`
- [ ] `@azure/msal-angularjs@1.x.x`

## Framework
React

## Description

This is the follow up to reported and closed [MSAL 1.4.0 was released, and is causing infinite redirection loop #2185](https://github.com/AzureAD/microsoft-authentication-library-for-js/issues/2185).

The redirection loop occurs, if the browser contains already cached values from previous MSAL versions. Manual clearing of applicaition cache helps to resolve the issue. But we cannot enforce all our users do this.

In my opinion, MSAL should be able to handle this case and not blow up on migration of the library.

## Error Message

**Cache error for scope openid,profile: The cache contains multiple tokens satisfying the requirements.**

hereby MSAL log (repeats as infinity loop):

```
M32 main.chunk.js:61058 MSAL: - Fri, 02 Oct 2020 07:24:54 GMT:1234-1.4.0-Verbose GetCachedToken has been called
VM32 main.chunk.js:61058 MSAL: - Fri, 02 Oct 2020 07:24:54 GMT:1234-1.4.0-Verbose No tokens found
VM32 main.chunk.js:61058 MSAL: - Fri, 02 Oct 2020 07:24:54 GMT:1234-1.4.0-Verbose Getting all cached access tokens
VM32 main.chunk.js:61058 MSAL: - Fri, 02 Oct 2020 07:24:54 GMT:1234-1.4.0-Verbose No authority passed, filtering tokens by scope
VM32 main.chunk.js:61058 MSAL: - Fri, 02 Oct 2020 07:24:54 GMT:1234-1.4.0-Info-pii multiple_matching_tokens:Cache error for scope openid,profile: The cache contains multiple tokens satisfying the requirements. Call AcquireToken again providing more requirements like authority..
VM32 main.chunk.js:61058 MSAL: - Fri, 02 Oct 2020 07:24:54 GMT:1234-1.4.0-Verbose AcquireTokenSilent has been called
VM32 main.chunk.js:61058 MSAL: - Fri, 02 Oct 2020 07:24:54 GMT:1234-1.4.0-Verbose Telemetry Event started: 4339d004-04e6-4b5a-ae89-7fd1d0f4835b_0e9d0ff7-f4ad-438d-a48f-0969873b6a2e-msal.api_event
VM32 main.chunk.js:61058 MSAL: - Fri, 02 Oct 2020 07:24:54 GMT:1234-1.4.0-Verbose-pii Serialized scopes: openid profile
VM32 main.chunk.js:61058 MSAL: - Fri, 02 Oct 2020 07:24:54 GMT:1234-1.4.0-Verbose Account set from MSAL Cache
VM32 main.chunk.js:61058 MSAL: - Fri, 02 Oct 2020 07:24:54 GMT:1234-1.4.0-Verbose Response type: id_token

```

## Regression
- [x] Did this behavior work before? 
Version: 1.3.6

## MSAL Configuration

see linked #2185 

## Reproduction steps

see linked #2185 

## Expected behavior

MSAL should be able to handle multiple cache entries on migration and prevent infinite redirection loop .

## Browsers/Environment
- [x] Chrome
- [x] Firefox
- [x] Edge
- [ ] Safari
- [ ] IE
- [ ] Other (Please add browser name here)

**Comment by kumarunster** [comment]
@hectormmg could you take a look please?

**Comment by Shigerua** [comment]
I have a similar issue. With the new 1.4.1 the login works once. After reloading my page I always get the error 'The cache contains multiple tokens satisfying the requirements. Call AcquireToken again providing more requirements'. 
Workaround is to clean the localstorage and sign in again. (Then it works for 1 time).

Never had this before. Downgrading msal works fine.

**Comment by hectormmg** [comment]
Hi @kumarunster. Thanks for following up, I'm sorry the fix in 1.4.1 didn't completely solve your issues. We're currently looking into this, which seems to be caused by caching the same tokens with different authorities. We'll follow up if we need more information, otherwise we'll get the fix out as soon as we can.

**Comment by graphicsxp** [comment]
just to confirm I'm experiencing the same issue after upgrading to 1.4.1.
I use it along with angular-msal so it took me a while to figure out where the issue was coming from until I stumbled upon this post.
